lib/logstash/outputs/elasticsearch.rb

```diff
@@ -263,6 +265,10 @@ class LogStash::Outputs::ElasticSearch < LogStash::Outputs::Base
   config :user, :validate => :string
   config :password, :validate => :password
 
+  # HTTP Path at which the Elasticsearch server lives. Use this if you must run ES behind a proxy that remaps
+  # the root path for the Elasticsearch HTTP API lives. This option is ignored for non-HTTP transports.
+  config :path, :validate => :string, :default => "/"
```

Comment:
I think we should be explicit in saying this option is invalid, not just ignored, for non-HTTP Transports. Further, we should reject a configuration of transport or nodeclient with a path setting on startup.